src/data_visualization/src/parse_cols.rs

```diff
@@ -0,0 +1,107 @@
+#![allow(dead_code)]
+#![allow(unused)]
+
+use std::{error::Error, fs::File, io::Read};
+
+use polars::{docs::lazy, prelude::*};
+
+use chrono::{DateTime, NaiveDate, Utc};
+
+use rand_distr::num_traits::Num;
+use std::str::FromStr;
+
+use rand;
+use rand::Rng;
+// use rand::Rng::gen_range;
+
+pub fn load_as_cols(filename: &str) -> PolarsResult<DataFrame> {
+	let file = File::open(filename)?;
+
+	let df = CsvReadOptions::default()
+		.with_has_header(true)
+		.try_into_reader_with_file_path(Some(filename.into()))?
+		.finish()?;
+
+	Ok(df)
+}
+
+pub fn transform_data(df: DataFrame) -> PolarsResult<DataFrame> {
```

Comment:
```suggestion
pub fn transform_data(mut df: DataFrame) -> PolarsResult<DataFrame> {
```

do not `clone` please